### PR TITLE
[ios] Fix testDeallocated failing locally. 

### PR DESCRIPTION
**Pull request by cyanglaz**
Moving the code block to autorelease pool to ensure the FlutterEngine is released in the test.

Fixes: https://github.com/flutter/flutter/issues/134388

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterEngineTest.mm [hunk 1]
```diff
@@ -113,7 +113,7 @@ - (void)testInfoPlist {
 
 - (void)testDeallocated {
   __weak FlutterEngine* weakEngine = nil;
-  {
+  @autoreleasepool {
```
Comment (gaaclarke) [inline shell/platform/darwin/ios/framework/Source/FlutterEngineTest.mm:116]: Were you able to track down how the engine is getting onto the autorelease pool?

**Review by gaaclarke — APPROVED** [review]
Approval since it demonstrably works.  I have some concerns that the issue is another thread that has a retain on the engine, not the autorelease pool.

File: shell/platform/darwin/ios/framework/Source/FlutterEngineTest.mm (hunk 1 above)
Comment (cyanglaz) [reply shell/platform/darwin/ios/framework/Source/FlutterEngineTest.mm:116]: No. I'm not sure how to do that. 

> I have some concerns that the issue is another thread that has a retain on the engine
In what situation could this happen?

Comment (cyanglaz) [reply shell/platform/darwin/ios/framework/Source/FlutterEngineTest.mm:116]: This is only happening when running the test locally tho, ci never failed. So Im not sure what the difference is that caused the FlutterEngine getting onto the pool.

**Pull request merged**